bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/types/StateDescriptionFragmentBuilder.java

```diff
@@ -0,0 +1,239 @@
+        this.minimum = minimum;
+        return this;
+    }
+
+    /**
+     * Set the step for the resulting {@link StateDescriptionFragment}.
+     *
+     * @param step the step for the resulting {@link StateDescriptionFragment}.
+     * @return this builder.
+     */
+    public StateDescriptionFragmentBuilder withStep(BigDecimal step) {
+        this.step = step;
+        return this;
+    }
+
+    /**
+     * Set the pattern for the resulting {@link StateDescriptionFragment}.
+     *
+     * @param pattern the pattern for the resulting {@link StateDescriptionFragment}.
+     * @return this builder.
+     */
+    public StateDescriptionFragmentBuilder withPattern(String pattern) {
+        this.pattern = pattern;
+        return this;
+    }
+
+    /**
+     * Set readOnly for the resulting {@link StateDescriptionFragment}.
+     *
+     * @param readOnly readOnly for the resulting {@link StateDescriptionFragment}.
+     * @return this builder.
+     */
+    public StateDescriptionFragmentBuilder withReadOnly(Boolean readOnly) {
+        this.readOnly = readOnly;
+        return this;
+    }
+
+    /**
+     * Set the {@link StateOption}s for the resulting {@link StateDescriptionFragment}.
+     *
+     * @param options the {@link StateOption}s for the resulting {@link StateDescriptionFragment}.
+     * @return this builder.
+     */
+    public StateDescriptionFragmentBuilder withOptions(List<StateOption> options) {
+        this.options = options;
+        return this;
+    }
+
+    /**
+     * Merge the given {@link StateDescriptionFragment}. Set all unset ({@code null}) fields of this builder to the
+     * values
+     * from the
+     * given {@link StateDescriptionFragment}.
+     *
+     * @param fragment a {@link StateDescriptionFragment} this builder should merge in.
+     * @return the builder.
+     */
+    public StateDescriptionFragmentBuilder mergeStateDescriptionFragment(StateDescriptionFragment fragment) {
+        if (this.minimum == null) {
+            this.minimum = fragment.getMinimum();
+        }
+        if (this.maximum == null) {
+            this.maximum = fragment.getMaximum();
+        }
+        if (this.step == null) {
+            this.step = fragment.getStep();
+        }
+        if (this.pattern == null) {
+            this.pattern = fragment.getPattern();
+        }
+        if (this.readOnly == null) {
+            this.readOnly = fragment.isReadOnly();
+        }
+        if (this.options == null) {
+            this.options = fragment.getOptions();
+        }
+
+        return this;
+    }
+
+    /**
+     * Merge the given {@link StateDescription}. Set all unset ({@code null}) fields of this builder to the values from
+     * the
+     * given {@link StateDescription}.
+     *
+     * @param legacy a {@link StateDescription} this builder should merge in.
+     * @return the builder.
+     */
+    public StateDescriptionFragmentBuilder mergeStateDescription(StateDescription legacy) {
```

Comment:
Wouldn't that rather be a canditate for a `public static ... create(StateDescription)` method? Or did I miss a use-case?